halibot/halauth.py

```diff
@@ -1,12 +1,27 @@
 import json
 import logging
 import re
+from halibot.message import Message, MalformedMsgException
 
-def hasPermission(perm, reply=False):
+def hasPermission(perm, reply=False, argnum=None, key="msg"):
 	def real_dec(func):
-		def wrapper(self, msg, *args, **kwargs):
+		def wrapper(self, *args, **kwargs):
+			if argnum != None and argnum < len(args):
+				msg = args[argnum]
+			else:
+				msg = kwargs.get(key)
```

Comment:
Maybe this should be `kwargs[key]`? If they provide the wrong key, it is probably better to point that out directly than to get a MalformedMsgException.

Reply:
Yeah, I think that resolves my question in the other thread. We should propagate the `KeyError` upwards.